src/test/ui/inference_unstable.stderr

```diff
@@ -0,0 +1,12 @@
+warning: a method with this name will be added to the standard library in the future
+  --> $DIR/inference_unstable.rs:26:20
+   |
+LL |     assert_eq!('x'.ipu_flatten(), 1);
+   |                    ^^^^^^^^^^^
+   |
+   = note: #[warn(unstable_name_collision)] on by default
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see pr #48552 <https://github.com/rust-lang/rust/pull/48552>
+   = help: call with fully qualified syntax `inference_unstable_itertools::IpuItertools::ipu_flatten(...)` to keep using the current method
+   = note: add #![feature(ipu_flatten)] to the crate attributes to enable `inference_unstable_iterator::IpuIterator::ipu_flatten`
```

Comment:
I definitely think this output is overwhelming. Let's customize the behavior of future-compat-warning for this particular lint. I think it should probably say something like this:

```
warning: a method with this name will be added to the standard library in the future
  --> $DIR/inference_unstable.rs:26:20
   |
LL |     assert_eq!('x'.ipu_flatten(), 1);
   |                    ^^^^^^^^^^^ method will be resolved differently in the future
   |
   = note: #[warn(unstable_name_collision)] on by default
   = warning: once this method is added to the standard library, this call will invoke the standard library method instead of this method from the crate `inference_unstable_itertools`
   = note: for more information, see pr #48552 <https://github.com/rust-lang/rust/pull/48552>
   = help: call with fully qualified syntax `inference_unstable_itertools::IpuItertools::ipu_flatten(...)` to keep using the current method
   = note: add #![feature(ipu_flatten)] to the crate attributes to enable `inference_unstable_iterator::IpuIterator::ipu_flatten`
```

@estebank may have input here too

Reply:
> warning: once this method is added to the standard library, this call will invoke the standard library method instead of this method from the crate `inference_unstable_itertools`

The two "this method" here refer to different methods, right? The first is in the standard library, the second to the one in a crate.

Reply:
I've rewritten the warning as

> once this method is added to the standard library, there will be ambiguity about which method to choose and will cause an error!

Reply:
"there will be X and will cause an error" sounds wrong to me -- is that just me not being a native speaker? "there will be ambiguity [...], which will cause an error" sounds better to me.

Reply:
@RalfJung

> once this method is added to the standard library, there will be ambiguity here, which will cause a hard error!

Better?

Reply:
Perfect :)